### PR TITLE
Add support for R languageserver

### LINKED ISSUE #3048: Support R languageserver
R has a full featured LSP client called languageserver.  After installing the package (outside the scope of ALE), it is called as follows:

R --slave -e "languageserver::run()"

Possible to add?

### PR DESCRIPTION
**Pull request by Dophin2009**
Resolves #3048 .

This PR adds support for [languageserver](https://github.com/REditorSupport/languageserver), the language server implementation for R.

### CODE REVIEW

**Comment by Dophin2009** [comment]
@w0rp Do you mind merging this?

**Comment by mutlusun** [comment]
Hello,
thanks for your work @Dophin2009 ! I tested this branch with neovim 0.4.3 and the languageserver implementation works as expected. Thank you!

Best

File: ale_linters/r/languageserver.vim [hunk 1]
```diff
@@ -0,0 +1,26 @@
+" Author: Eric Zhao <21zhaoe@protonmail.com>
+" Description: Implementation of the Language Server Protocol for R.
+
+call ale#Set('r_languageserver_cmd', 'languageserver::run()')
+call ale#Set('r_languageserver_options', {})
```
Comment (w0rp) [inline ale_linters/r/languageserver.vim:5]: I'll rename the option to `_config`. I like to use `_config` for LSP configuration, and `_options` for command line options.

**Review by w0rp — APPROVED** [review]

**Comment by zoj613** [comment]
which version of ALE is this feature getting released on?

**Comment by zoj613** [comment]
Thanks for this @Dophin2009 . I was just today looking to see if ALE supports this

**Pull request merged**